bin/satpy_launcher.py

```diff
@@ -31,19 +32,23 @@
 LOG = getLogger(__name__)
 
 
-def main():
-    # Product list is always the last argument
-    prod_list = sys.argv[-1]
-    if len(sys.argv) > 2:
-        # Collect all the topics, which can be either comma or space separated
-        topics = []
-        for arg in sys.argv[1:-1]:
-            topics += arg.split(',')
-    else:
-        topics = None
-
-    run(prod_list, topics=topics)
-
-
 if __name__ == "__main__":
-    main()
+
+    parser = argparse.ArgumentParser(
+        description='Launch trollflow2 processing with Satpy listening on the specified Posttroll topic(s)')
+    parser.add_argument("topics", nargs='*',
+                        help="The topic(s) - a space separated list of topics",
+                        type=str)
+    parser.add_argument("--product_list", '-p',
```

Comment:
I think product list filename should be kept as positional argument (without a command-line switch) so that command for operations would be a bit cleaner and wouldn't need to be changed.

Reply:
I don't really agree. Also, because I am not sure how to do that in a consistent/safe way?

Reply:
```python
parser.add_argument("config", help="config file to be used")```